components/engine_traits/src/mvcc_properties.rs

```diff
@@ -6,6 +6,33 @@ use txn_types::TimeStamp;
 
 use crate::TtlProperties;
 
+/// MVCC properties for estimating discardable versions based on GC safe point.
+///
+/// These statistics are collected to estimate the number of discardable MVCC
+/// versions using uniform distribution assumption. There are two types of
```

Comment:
The actual case can hardly be uniform distribution. Can we add some buckets to record how many stale version within the bucket?

Reply:
@hbisheng This is the potential project for the collaboration with our college partner. It can either be a simple solution, or a complex one. What do you think?